### PR TITLE
User model cleanup

### PR DESCRIPTION
**Pull request by stopfstedt**
just some cleanup, to prepare this model for the inevitable switching of Promises implementation.

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit d3b2c7768d9fd9efbbd8b33ea6368a62379fa941

https://deploy-preview-2772--ilios-frontend.netlify.com

File: app/models/user.js [hunk 1]
```diff
@@ -1,106 +1,105 @@
...
-  examined:  DS.attr('boolean'),
-  userSyncIgnore:  DS.attr('boolean'),
-  icsFeedKey:  DS.attr('string'),
-  root: DS.attr('boolean'),
-  reminders: DS.hasMany('user-made-reminder', {async: true}),
-  reports: DS.hasMany('report', {async: true}),
-  school: DS.belongsTo('school', {async: true}),
-  authentication: DS.belongsTo('authentication', {async: true}),
-  directedCourses: DS.hasMany('course', {
+const ProxyContent = Object.extend(PromiseProxyMixin);
```
Comment (jrjohnson) [inline app/models/user.js]: Can we get rid of this and just use a returned Promise where it is used?

File: app/models/user.js [hunk 2]
```diff
@@ -109,34 +108,32 @@ var User = DS.Model.extend({
...
         });
-        RSVP.all(promises).then(schools => {
+        all(promises).then(schools => {
           schools.pushObject(primarySchool);
-          defer.resolve(schools.uniq());
+          deferred.resolve(schools.uniq());
         });
       });
     });
     return PromiseArray.create({
```
Comment (jrjohnson) [inline app/models/user.js]: PromiseArray is a private API of ember-data - while cleaning this up I think switching to returning a plain Promise is probably a good idea.

**Comment by stopfstedt** [comment]
@jrjohnson thanks for the review. all of the above will be addressed in a second PR. i want to keep the change set small on this one, and just focus on cosmetics now before moving onto actual re-implementations.

**Pull request merged**